src/Inkluzitron/Modules/Vote/VotegagModule.cs

```diff
@@ -0,0 +1,210 @@
+        {
+            var minMinutes = VoteSettings.MuteMinutesMin;
+            var maxMinutes = VoteSettings.MuteMinutesMax;
+
+            if (minutes < minMinutes || minutes > maxMinutes)
+            {
+                await ReplyAsync($"Doba umlčení musí být v rozsahu {minMinutes}-{maxMinutes} minut.");
+                return;
+            }
+
+            if (Context.Guild.Id != BotSettings.HomeGuildId)
+            {
+                await ReplyAsync("Tento příkaz lze spustit jen na hlavním serveru.");
+                return;
+            }
+
+            if (user.IsBot)
+            {
+                await ReplyAsync("Nemůžeš umlčet bota.");
+                return;
+            }
+
+            var guildUser = Context.Guild.GetUser(user.Id);
+
+            if (guildUser == null)
+            {
+                await ReplyAsync($"{user.GetDisplayName()} není na tomto serveru.");
+                return;
+            }
+
+            var userDb = await UsersService.GetOrCreateUserDbEntityAsync(user);
+
+            var mutedGendered =
+                userDb.Gender == Gender.Male ? "umlčený" :
+                userDb.Gender == Gender.Female ? "umlčená" :
+                "umlčený/á";
+
+            var wasGendered =
+                userDb.Gender == Gender.Male ? "byl" :
+                userDb.Gender == Gender.Female ? "byla" :
+                "byl/a";
+
+            if (userDb.MutedUntil != null && userDb.MutedUntil > DateTime.Now)
+            {
+
+                await ReplyAsync($"{user.GetDisplayName()} již je {mutedGendered} do {userDb.MutedUntil?.ToString("HH:mm")}. Nelze tedy zahájit nové hlasování.");
+                return;
+            }
+
+            if(Cache.TryGetValue<DateTime>(CreateCacheKey(user.Id), out var voteEnd))
+            {
+                await ReplyAsync($"Již probíhá hlasování o umlčení {user.GetDisplayName()}. Do {voteEnd:HH:mm} tedy nelze zahájit nové hlasování.");
+                return;
+            }
+
+            var mentions = new AllowedMentions
+            {
+                MentionRepliedUser = false
+            };
+            mentions.UserIds.Add(guildUser.Id);
+
+            voteEnd = DateTime.Now.Add(VoteSettings.MuteVoteTimeout);
+            Cache.Set(CreateCacheKey(user.Id), voteEnd);
+
+            var minVotesCalculation = VoteSettings.MuteVoteMaxVotes -
+                Math.Pow((maxMinutes - minutes) / (double)(maxMinutes - minMinutes), 2)
+                * (VoteSettings.MuteVoteMaxVotes - VoteSettings.MuteVoteMinVotes);
+
+            var minVotes = (int)Math.Floor(minVotesCalculation);
+
+            var voteMessage = await ReplyAsync(GenerateVoteMessage(guildUser, minutes, minVotes, 0, voteEnd), allowedMentions:mentions);
+            await voteMessage.AddReactionAsync(VoteSettings.MuteReactionFor);
+            await voteMessage.AddReactionAsync(VoteSettings.MuteReactionAgainst);
+            await Task.Delay(VoteSettings.MuteVoteTimeout);
```

Comment:
to je hnus velebnosti

Reply:
i get it, ale daleko hezčí řešení by byla nějaká SchedulerService, která by tikala a které bys řekl přes .Schedule(DateTime, Action), ať ti v nějaký čas pustí tu action

Reply:
bonus points by ta service mohla ukládat ten svůj plán do databáze, pak by se tam samozřejmě necpala Action ale třeba nějaký JSON, a z VotegagModule by ses té službě nějak přihlásil k odběru určitého typu ("VoteGag") událostí, ale tohle už je dost advanced

Reply:
existuje i třeba [Quartz.NET](https://andrewlock.net/using-quartz-net-with-asp-net-core-and-worker-services/) knihovna která by spoustu z toho zvládla sama, ale nemám zkušenost, jak na tom fungujou jednorázové "joby"

Reply:
A ještě by to chtělo nějakou generickou službu, která bude tyhle tasky zpracovávat na základě nějaké pokročilé AI postavené na neuronových sítích :rtzW:

Reply:
Podle mě je tohle relativně good.